### PR TITLE
remove optimization_level=None from randomized tests

### PR DESCRIPTION
**Pull request by ajavadia**
since #2672, the default optimization level is just level 1, so there is no need to test `optimization_level=None` anymore.

### CODE REVIEW

File: test/randomized/test_transpiler_equivalence.py [hunk 1]
```diff
@@ -199,7 +199,6 @@ def qasm(self):
             st.none(),
             st.sampled_from(mock_backends)),
         opt_level=st.one_of(
```
Comment (kdk) [inline test/randomized/test_transpiler_equivalence.py]: This could be reduced to `opt_level=st.integers(min_value=0, max_value=3)`, but this is also fine as is.
